python/lsst/summit/extras/fastStarTrackerAnalysis.py

```diff
@@ -0,0 +1,625 @@
+    allSeqNums = set(s.seqNum for s in sources)
+    if len(allDayObs) > 1 or len(allSeqNums) > 1:
+        raise ValueError(
+            "The sources are from multiple days or sequences, found"
+            f" {allDayObs} dayObs and {allSeqNums} seqNum values."
+        )
+    dayObs = allDayObs.pop()
+    seqNum = allSeqNums.pop()
+    startFrame = min(frameNums)
+    endFrame = max(frameNums)
+
+    title = f"dayObs {dayObs}, seqNum {seqNum}, frames {startFrame}-{endFrame}"
+
+    axisLabelSize = 18
+
+    figs = []
+    fig = plt.figure(figsize=(10, 16))
+    ax1, ax2, ax3 = fig.subplots(3, sharex=True)
+    fig.subplots_adjust(hspace=0)
+
+    ax1.plot(frameNums, [s.rawFlux for s in sources], label="Raw Flux", **opts)
+    ax1.plot(frameNums, [s.hsmFittedFlux for s in sources], label="Fitted Flux", **opts)
+    ax1.set_ylabel("Flux (ADU)", size=axisLabelSize)
+    ax1.set_title(title)
+    ax1.legend()
+
+    ax2.plot(frameNums, [s.centroidX for s in sources], label="Raw centroid x", **opts)
+    ax2.plot(
+        frameNums,
+        [s.hsmCentroidX for s in sources],
+        label="Fitted centroid x",
+        **opts,
+    )
+    ax2.set_ylabel("x-centroid (pixels)", size=axisLabelSize)
+    ax2.legend()
+
+    ax3.plot(frameNums, [s.centroidY for s in sources], label="Raw centroid y", **opts)
+    ax3.plot(
+        frameNums,
+        [s.hsmCentroidY for s in sources],
+        label="Fitted centroid y",
+        **opts,
+    )
+    ax3.set_ylabel("y-centroid (pixels)", size=axisLabelSize)
+    ax3.set_xlabel("Frame number", size=axisLabelSize)
+    ax3.legend()
+
+    figs.append(fig)
+
+    fig = plt.figure(figsize=(10, 10))
+    ax4 = fig.subplots(1)
+
+    colors = np.arange(len(sources))
+    # gnuplot2 has a nice balance of nothing white, and having an intuitive
+    # progression of colours so the eye can pick out trends on the point cloud.
+    axRef = ax4.scatter(
+        [s.centroidX for s in sources],
+        [s.centroidY for s in sources],
+        c=colors,
+        cmap="gnuplot2",
+    )
+    ax4.set_xlabel("x-centroid (pixels)", size=axisLabelSize)
+    ax4.set_ylabel("y-centroid (pixels)", size=axisLabelSize)
+    ax4.set_aspect("equal", "box")
+    # move the colorbar
+    divider = make_axes_locatable(ax4)
+    cax = divider.append_axes("right", size="5%", pad=0.05)
+    cbar = plt.colorbar(axRef, cax=cax)
+    ax4.set_title(title)
+    cbar.set_label("Frame number in series", size=axisLabelSize * 0.75)
+    figs.append(fig)
+
+    return figs
+
+
+# -------------- plotting tools
+
+
+def plotSourcesOnImage(parentFilename, sources):
+    """Plot one of more source on top of an image.
+
+    Parameters
+    ----------
+    parentFilename : `str`
+        The full path to the parent (.tif) file.
+    sources : `list` of
+              `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source` or
+              `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source`
```

Comment:
`lsst.summit.utils.astrometry.fastStarTrackerAnalysis` -> `lsst.summit.utils.starTracker` ?

Reply:
Needs changing, but to `lsst.summit.extras.fastStarTrackerAnalysis.Source` I think.